go/pkg/cas/upload.go

```diff
@@ -108,14 +114,53 @@ type UploadOptions struct {
 // result in a dangling symlink.
 var ErrSkip = errors.New("skip file")
 
+// UploadResult is the result of a Client.Upload call.
+// It provides file/dir digests and statistics.
+type UploadResult struct {
+	Stats TransferStats
+	u     *uploader
+}
+
+// Digest returns the digest computed for a file/dir at ps.Path.
+//
+// To retrieve a digest of a regular file, only ps.Path is required.
+//
+// To retrieve a digest of a directory or a symlink, ps.Exclude must match one
```

Comment:
All of ps should match one that was previously uploaded, right? I think in general the passed `ps` should match one that was passed to upload, regardless of whether its a file or a directory? That would make the comment simpler?

Reply:
I am pretty sure I responded to this, but I cannot find it now...

>  I think in general the passed ps should match one that was passed to upload, regardless of whether its a file or a directory? 

no necessarily. Suppose `&PathSpec{ Path: "/foo/"}` was passed to `Client.Upload()`. The client may call `UploadResult.Digest(&PathSpec{ Path: "/foo/bar"})` although they never called `Upload` with that pathSpec.

Consider a more nuanced case.

```go
// For each file, get a directory containing that file also "/foo/always".
for _, f := range files {
  pathC <- &PathSpec{ Path: "/foo/", Allowlist: []string{f, "/foo/always"}}
}
```
Now suppose that the client, for whatever reason, wants digest of "/foo/always". It is simpler to call `UploadResult.Digest(&PathSpec{ Path: "/foo/always"})` than savings on of the previously-passed allowlist and then using that. The digest of "/foo/always" does not depend on the allowlist.

WDYT?

> That would make the comment simpler?

I agree that it would, but at the cost of more code, which seems like a higher cost.

Reply:
Ack.